base/intfuncs.jl

```diff
@@ -369,8 +369,19 @@ const HWNumber = Union{HWReal, Complex{<:HWReal}, Rational{<:HWReal}}
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{1}) = x
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{2}) = x*x
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{3}) = x*x*x
-@inline literal_pow(::typeof(^), x::HWNumber, ::Val{-1}) = inv(x)
-@inline literal_pow(::typeof(^), x::HWNumber, ::Val{-2}) = (i=inv(x); i*i)
+@inline function literal_pow(::typeof(^), x::HWNumber, ::Val{-1})
+    if x == -one(x)
```

Comment:
This change introduces even more type instability, now depending on the _value_ of the `x`  argument.

Other question: why special-case `-1` and not `1`?

Reply:
the reason to specialize only on -1 is that inv(x) often has a different type than x, but x*x generally is ok. what is a fade where this version is type unstable? unitful?

Reply:
I meant depending on the first argument: `typeof ((-1)^-1) == Int`, while `typeof(1^-1) == Float64)`